dao/src/main/java/org/thingsboard/server/dao/user/UserServiceImpl.java

```diff
@@ -194,11 +194,13 @@ public UserCredentials requestPasswordReset(TenantId tenantId, String email) {
         DataValidator.validateEmail(email);
         User user = userDao.findByEmail(tenantId, email);
         if (user == null) {
-            throw new IncorrectParameterException(String.format("Unable to find user by email [%s]", email));
+            log.error(String.format("Unable to find user by email [%s]", email));
```

Comment:
Why not throw an appropriate exception? Return null in such a case is bad practice.